tests/test_frameworks.py

```diff
@@ -153,13 +153,59 @@ def test_tensorflow(batch_size, shuffle_config, data=DATA, hidden_units=128):
     assert err < 1e-3
 
 
+@pytest.mark.skipif("not python_version_ok()", reason="need at least python 3.7")
+@pytest.mark.parametrize("batch_size,shuffle_config", [(1, 0), (32, 0), (32, 1), (32, 2)])
+def test_tensorflow_functional(batch_size, shuffle_config, data=DATA, hidden_units=128):
+    dataset_tf = tf.data.Dataset.from_tensor_slices((data["X"], data["y"]))
+    if shuffle_config == 0:  # proper shuffling for SGD
+        dataset_shuffled = dataset_tf.shuffle(buffer_size=len(data["X"]))
+    elif shuffle_config == 1:  # shuffling for datasets that don't fit in memory
+        dataset_shuffled = dataset_tf.shuffle(buffer_size=60)
+    else:
+        dataset_shuffled = dataset_tf  # no shuffling
+
+    dataset_og_order = dataset_tf.batch(batch_size)
+    dataset_tf = dataset_shuffled.batch(batch_size)
+
+    def make_model(num_features, num_classes):
+        inputs = tf.keras.Input(shape=(num_features,))
+        x = tf.keras.layers.Dense(64, activation="relu")(inputs)
+        outputs = tf.keras.layers.Dense(num_classes)(x)
+        model = tf.keras.Model(inputs=inputs, outputs=outputs, name="test_model")
+
+        return model
+
+    model = KerasWrapperModel(
+        make_model,
+        model_kwargs={"num_features": data["num_features"], "num_classes": data["num_classes"]},
+    )
+
+    # Test base model works:
+    model.fit(
+        X=dataset_tf,
+        y=data["y"],
+        epochs=2,
+    )
+    preds_base = model.predict_proba(dataset_tf)
+
+    # Test CleanLearning performs well:
+    cl = CleanLearning(model)
+    cl.fit(dataset_tf, data["y"], clf_kwargs={"epochs": 10}, clf_final_kwargs={"epochs": 15})
+
+    preds = cl.predict(dataset_og_order)
+    err = np.sum(preds != data["y_og"]) / len(data["y_og"])
+    issue_indices = list(cl.label_issues_df[cl.label_issues_df["is_label_issue"]].index.values)
+    assert set(issue_indices).issubset(set(data["error_indices"]))
```

Comment:
might this check start to fail if Keras releases new version with eg. different weight-initializations / optimizer (ie. how robust do you think it is)?  would it be more future proof if we instead check their intersection is nonempty or Jaccard similarity is at least `2 / len(data["error_indices"]` or something like that which doesn't require strict subset?

Reply:
Updated to check for non-empty intersection